R/pca.R

```diff
@@ -155,6 +159,8 @@ step_pca_new <-
 
 #' @export
 prep.step_pca <- function(x, training, info = NULL, ...) {
+  x <- add_new_args(x)
```

Comment:
I think the way this works is a bit too complicated.

Maybe we could just use a check for `is.null(x$keep_vars)` in the `prep()` and `bake()` methods for these steps, and set it to `FALSE` if it is NULL, but then throw a warning that the user should regenerate their recipe because the arguments have been updated?

Reply:
Yeah, that's what I did at first (with %||%), but I had the beginnings of a more reusable idea walking back from dev day and wanted to give it a shot. I agree it's probably a bit too fragile right now, but wanted to log the general idea to get gears turning for a prep/bake backward compatibility safety valve. I'll go back to the other method.

Reply:
Ok, I rolled back the over-complicated change and added a warning. Can you review the `step_pca` implementation to make sure everything looks in-order? If so, I'll apply it to the other 6 functions where it makes sense.